### PR TITLE
upstream: respect overprovisioning factor for static/strict dns

### PR DESCRIPTION
**Pull request by snowp**
Plumbs through the value set for the overprovisioning factor for
STATIC/STRICT_DNS. Previously these values would be ignored.

This came up as part of looking into #5089 

Signed-off-by: Snow Pettersen <snowp@squareup.com>

*Risk Level*: Medium, might affect existing config
*Testing*: UTs
*Docs Changes*: n/a
*Release Notes*: n/a

### CODE REVIEW

**Comment by snowp** [comment]
@dio Which part of the documentation would you want me to add the example to?

**Comment by dio** [comment]
@snowp in relation with https://github.com/envoyproxy/envoy/pull/5091/files#diff-5560832ea7741e9a4298f2f4d4dfdae5R926 I think will be helpful to put in here: https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/eds.proto.html?highlight=overprovisioning#clusterloadassignment-policy at the overprovisioning_factor section.

File: api/envoy/api/v2/eds.proto [hunk 1]
```diff
@@ -89,7 +89,12 @@ message ClusterLoadAssignment {
     // multiplied by the overprovisioning factor drops below 100.
     // With the default value 140(1.4), Envoy doesn't consider a priority level
     // or a locality unhealthy until their percentage of healthy hosts drops
-    // below 72%.
+    // below 72%. For example:
+    //
+    // .. code-blocK:: json
```
Comment (mattklein123) [inline api/envoy/api/v2/eds.proto]: typo: "blocK" (maybe it's case-insensitive but might want to check rendered docs or just fix to lower case)

**Pull request merged**